src/components/Attachments/AttachmentCarousel/index.tsx

```diff
@@ -51,6 +51,9 @@ function AttachmentCarousel({report, reportActions, parentReportActions, source,
         const attachmentsFromReport = extractAttachmentsFromReport(parentReportAction, reportActions ?? undefined);
 
         if (isEqual(attachments, attachmentsFromReport)) {
+            if (attachments.length === 0) {
+                setPage(-1);
+            }
             return;
```

Comment:
We need to make a little refactor here as we also need to call `setDownloadButtonVisibility` to hide the download button.

Reply:
Actually it may be better to remove this early return and remove the unnecessary dependencies same as in index.native.ts

Reply:
> we also need to call setDownloadButtonVisibility to hide the download button.

Oh, I didn't notice this one. Updated.

> Actually it may be better to remove this early return and remove the unnecessary dependencies same as in index.native.ts

I thought about this one before too, but it was added in the video player PR https://github.com/Expensify/App/pull/30829. I can't find a discussion for that though, so not sure if it's safe to change the deps.